api-tests/checkout-tests/npg/checkout-for-ecommerce-api.tests.json

```diff
@@ -1075,19 +1075,30 @@
 					"listen": "test",
 					"script": {
 						"exec": [
-							"pm.test(\"[eCommerce for checkout] All payment methods successfully retrieved with GET payment-methods/:methodId\", function(){",
+							"pm.test(\"[eCommerce for checkout] All payment methods successfully retrieved with GET payment-methods/:methodId\", function () {",
 							"    pm.response.to.have.status(200);",
 							"",
 							"    const response = pm.response.json();",
 							"",
-							"    const expectedPaymentMehtodKeys = new Set([\"asset\",\"description\",\"id\",\"name\",\"paymentTypeCode\",\"ranges\",\"status\",\"methodManagement\"]);",
+							"    const expectedPaymentMehtodKeys = new Set([\"asset\", \"description\", \"id\", \"name\", \"paymentTypeCode\", \"ranges\", \"status\", \"methodManagement\"]);",
+							"    const paymentMethodsWithoutBrandAssets = response.paymentMethods.map(p => {",
+							"        const { brandAssets, ...rest } = p;",
+							"        return rest;",
+							"    });",
+							"    const cardsPaymentMethods = response.paymentMethods.filter(p => p.paymentTypeCode == \"CP\");",
 							"    //this line handle transient time for which response will differs from DEV and UAT until Checkout develop branch deploy in UAT env",
 							"    const paymentMethods = response.paymentMethods == null ? response : response.paymentMethods;",
 							"    pm.expect(paymentMethods).to.be.not.empty;",
-							"    pm.expect(paymentMethods).to.satisfy((methods) => methods.every(method => Object.keys(method).every(property => expectedPaymentMehtodKeys.has(property))));",
+							"    pm.expect(paymentMethods).to.satisfy((methods) => paymentMethodsWithoutBrandAssets.every(method => Object.keys(method).every(property => expectedPaymentMehtodKeys.has(property))));",
```

Comment:
Shouldn't this check be on all methods instead on only methods without brand assets?

Reply:
Brand assets will be returned only for cards payment methods and not for others (since other methods does not have brand logo specialization, by the moment).
This check is performed against all payment methods filtering out the optional `brandAssets` field and checking that all the other fields are present then perform specific check against brandAssets field value only for the payment methods that have this property set